buffer/src/main/java/io/netty5/buffer/DefaultCompositeBuffer.java

```diff
@@ -574,6 +574,34 @@ public int transferTo(WritableByteChannel channel, int length) throws IOExceptio
         return totalBytesWritten;
     }
 
+    @Override
+    public int transferTo(FileChannel channel, long position, int length) throws IOException {
+        if (!isAccessible()) {
+            throw bufferIsClosed(this);
+        }
+        length = Math.min(readableBytes(), length);
+        if (length == 0) {
+            return 0;
+        }
+        checkReadBounds(readerOffset(), length);
+        int totalBytesWritten = 0;
+        try (var iterator = forEachComponent()) {
+            ByteBuffer[] byteBuffers = new ByteBuffer[countReadableComponents()];
+            int counter = 0;
+            for (var c = iterator.firstReadable(); c != null; c = c.nextReadable()) {
+                byteBuffers[counter++] = c.readableBuffer();
+            }
+            int bufferCount = countAndPrepareBuffersForChannelIO(length, byteBuffers);
+            for (int i = 0; i < bufferCount; i++) {
+                int bytesWritten = channel.write(byteBuffers[i], addExact(position, totalBytesWritten));
+                totalBytesWritten = addExact(totalBytesWritten, bytesWritten);
```

Comment:
This looks suspicious, both here and in the code you copied. We should either exit the loop and return the total, or avoid advancing `i`, if `byteBuffers[I].remaining() != 0`.

Reply:
good point, i will try to make a test

Reply:
fixed and added a test, though it's a bit ugly for FileChannel.